src/select.c

```diff
@@ -441,30 +441,23 @@ static __inline int listen_queued (Socket *socket)
  * remaining life of the socket.
  */
 #define READ_STATE_MASK   (SS_CANTRCVMORE)  /* set in recv() or shutdown() */
-#define WRITE_STATE_MASK  (SS_CANTSENDMORE | SS_ISCONNECTED)
+#define WRITE_STATE_MASK  (SS_CANTSENDMORE)
 
 static __inline int sock_signalled (Socket *socket, int mask)
 {
   if (socket->so_state & mask)
      return (1);
 
-  /* A normal blocking socket. 'so_error' is set in
-   * connect(), a "ICMP Unreachable" callback or when RST received
-   * in pctcp.c. Otherwise 'so_error' is zero.
-   */
-  if (!(socket->so_state & SS_NBIO))
-     return (socket->so_error);
+  if (socket->so_state & SS_NBIO)
+  {
+    if (socket->so_options & SO_ACCEPTCONN) /* non-blocking listen sock */
+       return (0);
```

Comment:
Should all errors be ignored on a listen socket?

Reply:
Not sure. In what scenario should a non-blocking listen socket return an error?

Reply:
Right, it doesn't matter, because there aren't any.  I guess you could call `shutdown()` on it but that doesn't make much sense.